### DISCUSSION TITLE
How  to autopopulate author/user relationships

### DISCUSSION THREAD

**Discussion by stratosgear** [pocketbase/pocketbase#5137] [Q&A]
New Pocketbase user here, testing the waters.

BTW, I already searched in the discussions, could find anything related...

I have a usecase (multiple actually) where I want to create rows in a Settings table having two columns (apart from the standard id, and date related):

* `user` relationship:Users single
* `data` json

Obviously this requires the user to be logged in.

Also, obviously, I would assume that there would be a way for me to automatically populate the `user` field from the @request.auth.id property, or that Pocketbase would automatically populate it if missing, but I do not see any way of doing it.  The information is already there in the incoming token, why would I need to also push the user id from the frontend as a separate field?  Do I have to use custom hooks for that?

There are multiple other tables where a user relationship like this would be required, so this might be too much boilerplate if I cannot somehow "automate" it.

Did I miss something obvious?

My suggestion would be some kind of Admin interface gui widget, that would suggest a default value of @request.auth.id, during the field creating while adding a New Collection, if the user adds a single User relationship.

Thanks!!!

**Comment by elsbrock** [comment]
The only way I know about is using the OnRecordBeforeCreateRequest or OnRecordBeforeUpdateRequest hooks. Here's an example for go:

```go
app.OnRecordBeforeCreateRequest("mycollection").Add(func(e *core.RecordCreateEvent) error {
	log.Println("RecordCreateEvent")
	authRecord, _ := e.HttpContext.Get(apis.ContextAuthRecordKey).(*models.Record)
	if authRecord == nil {
		return apis.NewForbiddenError("Only auth records can access this endpoint", nil)
	}

	e.Record.Set("user", authRecord.GetId())

	return nil
})

app.OnRecordBeforeUpdateRequest("mycollection").Add(func(e *core.RecordUpdateEvent) error {
	log.Println("RecordUpdateEvent")
	authRecord, _ := e.HttpContext.Get(apis.ContextAuthRecordKey).(*models.Record)
	if authRecord == nil {
		return apis.NewForbiddenError("Only auth records can access this endpoint", nil)
	}

	e.Record.Set("user", authRecord.GetId())

	return nil
})
```

One thing I didn't figure out yet is that the API still expects the fields to be provided.

**Reply by jonshipman** [reply]
You provide a dummy user id in the request.

**Reply by elsbrock** [reply]
That’s what I’m doing right now. Is that the convention?

**Reply by ganigeorgiev** [reply]
@elsbrock It is not the "convention" but there is no other option at the moment if you want the "Nonempty" constraint because the hook is checked after the the validation. This behavior already has changed as part of the [ongoing refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) and the new hooks execution order.

**Reply by elsbrock** [reply]
Fantastic, that's what I needed to know. Thank you!

**Reply by TonyDMorris** [reply]
not sure if this is useful but I was working through the same problem and have solved it without requiring beforeUpdate hooks using the following api rules 
<img width="683" alt="image" src="https://github.com/user-attachments/assets/08738905-3216-4199-ae5b-0e06bd827909">

you still have to provide the userId but it has to match the auth user id and the record cannot be changed from one user to another
